### ISSUE TITLE
BAR crashes instead of shutting down gracefully - only the first time.

### ISSUE THREAD

**Issue by BitcoinMechanic** [Start9Labs/burn-after-reading#12]
![Screen Shot 2021-11-24 at 8 41 35 PM](https://user-images.githubusercontent.com/45926711/143381037-3fd02494-f653-47cd-851e-c0d977caea14.png)

**Comment by ProofOfKeags** [comment]
Are you saying subsequent shutdowns work fine?

**Comment by dr-bonez** [comment]
They might all be crashing, but the crash notification is being debounced.

**Comment by BitcoinMechanic** [comment]
@ProofOfKeags yes

**Comment by chrisguida** [comment]
Getting an internal compiler error when I try to build.

@elvece have you seen this?

```
docker run --rm -it -v ~/.cargo/registry:/root/.cargo/registry -v "/home/cguida/work/burn-after-reading"/backend:/home/rust/src start9/rust-musl-cross:aarch64-musl cargo +beta build --bins --release
   Compiling warp v0.2.5
error: internal compiler error: unexpected concrete region in borrowck: ReStatic
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.2.5/src/filters/sse.rs:457:5
    |
457 | /     pub fn stream<S>(
458 | |         self,
459 | |         event_stream: S,
460 | |     ) -> impl TryStream<
...   |
476 | |         }
477 | |     }
    | |_____^
    |
    = note: delayed at compiler/rustc_mir/src/borrow_check/region_infer/opaque_types.rs:87:44

error: internal compiler error: unexpected concrete region in borrowck: ReStatic
   --> /root/.cargo/registry/src/github.com-1ecc6299db9ec823/warp-0.2.5/src/filters/sse.rs:492:1
    |
492 | / pub fn keep<S>(
493 | |     event_stream: S,
494 | |     keep_interval: impl Into<Option<Duration>>,
495 | | ) -> impl TryStream<
...   |
514 | |     }
515 | | }
    | |_^
    |
    = note: delayed at compiler/rustc_mir/src/borrow_check/region_infer/opaque_types.rs:87:44

thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', compiler/rustc_errors/src/lib.rs:1050:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic


```

**Comment by chrisguida** [comment]
Fixed by 9e08705.  @BitcoinMechanic please test

**Comment by BitcoinMechanic** [comment]
Tested - works.
